### PR TITLE
chore(ui): Create config for ui-component CI job

### PR DESCRIPTION
**Pull request by dvail**
## Description

Adds job configuration to run Cypress component tests via Prow. This job is currently optional and runs only when invoked explicitly.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Kick off prow CI job manually with `/test ui-component-tests`.

Tests complete successfully and:

JUnit output successfully reported in Prow dashboard
image

JUnit reports uploaded to GCP
image

Cypress test videos captured and uploaded to GCP
image
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #9140** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/9140?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/9140?utm_source=stack-comment).

File: ui/apps/platform/scripts/cypress-component.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+artifacts_dir="${TEST_RESULTS_OUTPUT_DIR:-cypress/test-results}/artifacts"
+export CYPRESS_VIDEOS_FOLDER="${artifacts_dir}/videos"
+export CYPRESS_SCREENSHOTS_FOLDER="${artifacts_dir}/screenshots"
+
+DEBUG="*" NO_COLOR=1 cypress "$@" 2> /dev/null
```
Comment (gavin-stackrox) [inline ui/apps/platform/scripts/cypress-component.sh:6]: It feels odd to discard stderr (though I see this in the other script). I'd save it.
```suggestion
DEBUG="*" NO_COLOR=1 cypress "$@" 2> "${ARTIFACT_DIR:-/tmp}/cypress-err.txt"
```

File: Makefile [hunk 2]
```diff
@@ -549,6 +549,10 @@ endif
 ui-test:
 	make -C ui test
 
+.PHONY: ui-test-component
+ui-test-component:
+	make -C ui test-component
```
Comment (gavin-stackrox) [inline Makefile:554]: To match the job name.
```suggestion
.PHONY: ui-component-tests
ui-component-tests:
	make -C ui test-component
```

File: scripts/ci/jobs/ui-component-tests.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../../.. && pwd)"
+# shellcheck source=../../../scripts/ci/lib.sh
+source "$ROOT/scripts/ci/lib.sh"
+
+set -euo pipefail
+
+ui_component_tests() {
+    info "Starting UI component tests"
+    make ui-test-component || touch FAIL
```
Comment (gavin-stackrox) [inline scripts/ci/jobs/ui-component-tests.sh:11]:
```suggestion

    make ui-component-tests || touch FAIL
```

**Review by gavin-stackrox — APPROVED** [review]
LGTM with some suggestions.

**Pull request merged**